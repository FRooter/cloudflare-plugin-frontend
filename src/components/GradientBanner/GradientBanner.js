import React from 'react';
import { createComponent } from 'cf-style-container';

const gradientStyles = () => ({
  width: '100%',
  height: '150px',
  display: 'flex',
  justifyContent: 'center',
  background: 'linear-gradient(left, #8176B5, #76C4E2)',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '100% 250px',
  backgroundColor: '#e6e6e6'
});

const GradientBanner = ({ className, children }) => {
  return React.createElement('div', { className }, children);
};

export default createComponent(gradientStyles, GradientBanner);
