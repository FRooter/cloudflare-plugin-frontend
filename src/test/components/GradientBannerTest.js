import React from 'react';
import GradientBanner from '../../components/GradientBanner/GradientBanner';
import renderer from 'react-test-renderer';
import { StyleProvider } from 'cf-style-provider';

it('renders enabled correctly', () => {
  const component = renderer
    .create(
      <StyleProvider
        cssNode={null}
        fontNode={null}
        dev={false}
        selectorPrefix="cf-"
      >
        <GradientBanner
          imgSrc="imgSrc"
          title="title"
          description="description"
        />
      </StyleProvider>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
