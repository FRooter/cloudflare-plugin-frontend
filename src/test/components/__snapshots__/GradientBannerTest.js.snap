// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders enabled correctly 1`] = `
<div
  className="cf-a cf-b cf-c cf-d cf-e cf-f cf-g cf-h"
/>
`;
